spec/models/course/discussion/topic_spec.rb

```diff
@@ -83,5 +83,40 @@
         end
       end
     end
+
+    describe '.annotations_and_comments' do
+      let(:course) { create(:course) }
+      let!(:annotation) do
+        annotation =
+          create(:course_assessment_answer_programming_file_annotation, course: course).acting_as
+        create(:course_discussion_post, topic: annotation)
```

Comment:
Should we create a trait in the `:course_discussion_topic` factory? 

```ruby
factory :course_discussion_topic, class: Course::Discussion::Topic.name do
  ...
  trait :with_post do
    association :post, factory: :course_discussion_post
  end
  ...
end
```

Then we can pass the trait `:with_post` to the `:course_assessment_answer_programming_file_annotation` factory which will do the same for `:course_discussion_topic`?